### PR TITLE
Expire more mpl3.6 deprecations.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Review by ksunden — APPROVED** [review]

**Review by oscargus — APPROVED** [review]
Looks good! I'll let @QuLogic decide if he thinks that the removal notes should be organized in a different way (like separated on module) as he will make use of them later.

File: lib/matplotlib/backends/backend_wx.py [hunk 1]
```diff
@@ -931,28 +915,6 @@ def __init__(self, num, fig, *, canvas_class):
...
-        "3.6", alternative="frame.canvas.manager.toolmanager")(
-            property(lambda self: self.canvas.manager.toolmanager))
-
-    @_api.deprecated(
-        "3.6", alternative="the canvas_class constructor parameter")
-    def get_canvas(self, fig):
-        return FigureCanvasWx(self, -1, fig)
-
-    @_api.deprecated("3.6", alternative="frame.canvas.manager")
-    def get_figure_manager(self):
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_wx.py:952]: This appears to be missing from the docs; might be typo'd as `get_figure`?

File: lib/matplotlib/markers.py [hunk 2]
```diff
@@ -225,8 +224,8 @@ class MarkerStyle:
 
     _unset = object()  # For deprecation of MarkerStyle(<noargs>).
```
Comment (QuLogic) [inline lib/matplotlib/markers.py:225]: Delete as well.

File: lib/matplotlib/tests/test_colorbar.py [hunk 3]
```diff
@@ -322,11 +321,8 @@ def test_colorbarbase():
 
 
 def test_parentless_mappable():
-    pc = mpl.collections.PatchCollection([], cmap=plt.get_cmap('viridis'))
-    pc.set_array([])
-
-    with pytest.warns(_api.MatplotlibDeprecationWarning,
-                      match='Unable to determine Axes to steal'):
+    pc = mpl.collections.PatchCollection([], cmap=plt.get_cmap('viridis'), array=[])
+    with pytest.raises(ValueError):
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_colorbar.py:325]: Should keep the `match` on here.
```suggestion
    with pytest.raises(ValueError, match='Unable to determine Axes to steal'):
```

**Comment by anntzer** [comment]
Thanks, all fixed.

**Pull request merged**